docs/migrations/chain-upgrade-guide-040.md

```diff
@@ -0,0 +1,228 @@
+being slashed. The most important piece of this procedure is verifying your software version and genesis file hash before
+starting your validator and signing.
+
+The riskiest thing a validator can do is discover that they made a mistake and repeat the upgrade procedure again during
+the network startup. If you discover a mistake in the process, the best thing to do is wait for the network to start
+before correcting it. If the network is halted and you have started with a different genesis file than the expected one,
+seek advice from the validator community.
+
+### Recovery
+* Prior to exporting the state, the validators are encouraged to take a full data snapshot at exported height. Exported
+height will be determined by a governance proposal. Data backup is usually done by copying daemon home directory,
+ex: `~/.gaiad`
+
+It is critically important to back-up the validator state file, ex: `~/.gaiad/data/priv_validator_state.json` file
+after stopping your daemon process. This file is updated every block as your validator participates in a consensus
+rounds. It is a critical file needed to prevent double-signing, in case the upgrade fails and the previous chain needs
+to be restarted.
+
+In the event that the upgrade does not succeed, validators and operators must downgrade back to old version of the
+software and restore to their latest snapshot before restarting their nodes.
+
+### Upgrade procedure
+- Use old binary to export the state. Make sure to verify your binary version before exporting the state
+- Export the state from existing chain using old-binary (which uses `sdk@0.39.x`).
+Example:
+    ```sh
+    gaiad export --for-zero-height --height <height> > 039_exported_state.json
+    ```
+- Verify the SHA256 of the (sorted) exported genesis file:
+    ```shell
+    $ jq -S -c -M '' 039_exported_state.json | shasum -a 256
+    [PLACEHOLDER]  039_exported_state.json
+    ```
+- Cross check the hash with other peers (other validators) in the chat rooms
+- Install the latest binary (which uses `0.40`)
+- Migrate the exported state to `0.40` compatible genesis state
+    ```shell
+    gaiad migrate v40 039_exported_state.json --chain-id <new_chain_id> --genesis-time <new_genesis_time_in_utc> > new_v40_genesis.json
+    ```
+  NOTE: The migrate command takes an input genesis state and migrates it to a targeted version. New `genesis-time` will
+  be as mentioned in the governance proposal.
+- Verify the SHA256 of the migrated genesis file with other valdiators to make sure there are no manual errors in the process.
+    ```shell
+    $ jq -S -c -M '' new_v40_genesis.json | shasum -a 256
+    [PLACEHOLDER]  new_v40_genesis.json
+    ```
+- Make sure to update the genesis parameters in the new genesis if any. All these details will be generally present in
+the governance proposal
+- All the necessary state chanegs are handled in `040` migration command, including tendermint params. So, manual updates to
+the genesis are not required
```

Comment:
Is this true? tendermint params are not included in the sdk's migration commands.

Maybe an example would be ideal, using jq and updating max_age.

Reply:
I think it's true. We didn't add/update any params manually for bigbang upgrade. I will cross check and confirm. We just needed to add ibc related state, nothing else.

Reply:
Ah ok, so the tendermint consensus evidence params (max_age_num_blocks, max_age_duration) where already present in 0.39?

BTW, I think adding how to initialize IBC state would also be a good idea! "If your chain is using IBC, make sure to..."